### DISCUSSION TITLE
Choose inventory dynamically/on task run

### DISCUSSION THREAD

**Discussion by ahoiroman** [semaphoreui/semaphore#1415] [Q&A]
In ansible-semaphore, I would like to manage different servers using the API, which means that a app creates inventories and also adds secrets to the key store.

Now, here's my challenge: Each new host has it's own custom sudo secret and SSH key.

I want to define a Task Template, but I am required to specify an Inventory for it. This Inventory is associated with credentials (user/sudo). If I am not mistaken, I'd have to create a new task template for each host.

However, I don't want to do that. Instead, I want to use a Task Template for any host that has its own inventory. So, it would be great if I could dynamically choose the inventory and the associated credentials for a task. Is this possible?

**Comment by mapdeep** [comment]
I am also looking for a similar feature. maybe a dropdown to select the inventory before running a task.

**Comment by tjaartvdwalt** [comment]
I believe this was recently added. See #694
